### PR TITLE
chore: Configure automated documentation linting

### PR DESCRIPTION
**Pull request by beritou**
Configure linter to run in CI using GitHub Action. Fail build if linter detects any error-level problems.

Document how to run linter locally.

Apply default style provided by linter, along with the style from the write-good project.

Apply a custom Lightbend style inspired by the Microsoft Writing Style Guide https://docs.microsoft.com/en-us/style-guide/welcome/.

See README-vale.md for additional implementation and usage details.

Closes: lightbend/kalix#7488

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @beritou,

Thank you for your contribution! We really value the time you've taken to put this together.

We see that you have signed the Lightbend Contributors License Agreement before, however, the CLA has changed since you last signed it.
Please review the new CLA and sign it before we proceed with reviewing this pull request:

[https://www.lightbend.com/contribute/cla](https://www.lightbend.com/contribute/cla)

File: docs/src/modules/java/pages/call-another-service.adoc [hunk 1]
```diff
@@ -7,7 +7,7 @@ In some cases it is useful to call a component in another service, for example i
 
 Calling other Kalix services in the same project from an Action is done by invoking them over gRPC much like how an external client would. The service is however identified only by the name it has been deployed as, Kalix takes care of routing requests to the service and keeping the data safe by encrypting the connection for us.
 
-In this sample we will make an action that does two sequential calls to the xref:value-entity.adoc[Value Entity Counter] service, deployed with the service name "counter".
+In this sample we will make an action that does two sequential calls to the xref:value-entity.adoc[Value Entity Counter] service, deployed with the service name "counter."
```
Comment (efgpinto) [inline docs/src/modules/java/pages/call-another-service.adoc:10]: Interesting.. this is a new one for me. Always learning I guess :)

**Review by efgpinto — APPROVED** [review]
LGTM.. very useful 👍

File: docs/src/modules/java/pages/call-another-service.adoc (hunk 1 above)
Comment (pvlugter) [reply docs/src/modules/java/pages/call-another-service.adoc:10]: That looks odd. It's a name, not a quotation that should include punctuation inside the quotes. Maybe should be in backticks instead.

Comment (beritou) [reply docs/src/modules/java/pages/call-another-service.adoc:10]: I realize this discussion is mostly joking around, but if we disagree with any of these grammar rules, we can override the rule via configuration in https://github.com/lightbend/kalix-jvm-sdk/tree/main/styles

**Pull request merged**